ssl/statem/extensions_srvr.c

```diff
@@ -1151,6 +1151,7 @@ int tls_parse_ctos_psk(SSL *s, PACKET *pkt, unsigned int context, X509 *x,
             if (sesstmp == NULL) {
                 SSLfatal(s, SSL_AD_INTERNAL_ERROR,
                          SSL_F_TLS_PARSE_CTOS_PSK, ERR_R_INTERNAL_ERROR);
+                SSL_SESSION_free(sess);
```

Comment:
use this instead..
goto err:

Needs to go back to master..

Reply:
Changed.